### PR TITLE
chore: add more logs for OOM search

### PR DESCRIPTION
**Pull request by shoom3301**
This code doesn't change the logic, it only adds more logs and refactors the code a bit.

1. Added more logs
2. Removed excessive exports
3. Got rid of excessive ComposableCoW contract instances
4. Added "Conditional order params are equal but not the same" log

The last point is my suspect. I assume that comparation here is not valid, we should not use shallow comparison for the objects.

### CODE REVIEW

**Comment by shoom3301** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/domain/events/index.ts [hunk 1]
```diff
@@ -21,6 +22,9 @@ import { ConditionalOrder, ConditionalOrderParams } from "@cowprotocol/cow-sdk";
 
 import { ChainContext } from "../../services/chain";
 
+const composableCow = ComposableCoW__factory.createInterface();
+const log = getLogger("addContract:_addContract");
```
Comment (anxolin) [inline src/domain/events/index.ts:26]: Why is this log saying `addContract` if its just on top of this file which is called `events`?

File: src/domain/events/index.ts [hunk 2]
```diff
@@ -86,15 +90,16 @@ async function _addContract(
   }
 }
 
-export async function _registerNewOrder(
+async function registerNewOrder(
   event: ConditionalOrderCreatedEvent | MerkleRootSetEvent,
   composableCow: ComposableCoWInterface,
```
Comment (anxolin) [inline src/domain/events/index.ts:95]: Why do yo need to receive the `composableCow` if you already have it in scope?

File: src/domain/events/index.ts [hunk 3]
```diff
@@ -206,6 +212,23 @@ export function add(
     // Iterate over the conditionalOrders to make sure that the params are not already in the registry
     for (const conditionalOrder of conditionalOrders?.values() ?? []) {
       // Check if the params are in the conditionalOrder
+      if (conditionalOrder) {
```
Comment (anxolin) [inline src/domain/events/index.ts:215]: maybe add a TODO to delete these lines, and schedule the deletion. I understand you only wrote this to see how often it was happening we were adding an order when it was not time

File: src/domain/events/index.ts [hunk 4]
```diff
@@ -245,6 +268,7 @@ export function add(
         },
       ])
     );
+    // TODO: why twice?
```
Comment (anxolin) [inline src/domain/events/index.ts:271]: One increment owners and one orders

File: src/domain/events/index.ts [hunk 5]
```diff
@@ -256,19 +280,20 @@ export function add(
...
-          // Delete the conditional order
-          conditionalOrders.delete(conditionalOrder);
-        }
-      }
+function flush(owner: Owner, root: BytesLike, registry: Registry) {
+  if (!registry.ownerOrders.has(owner)) return;
+
+  const conditionalOrders = registry.ownerOrders.get(owner);
+
+  if (conditionalOrders === undefined) return;
```
Comment (anxolin) [inline src/domain/events/index.ts:288]: simpler, good call

File: src/types/model.ts [hunk 6]
```diff
@@ -95,27 +96,32 @@ export interface RegistryBlock {
   hash: string;
 }
 
+type OrdersPerOwner = Map<Owner, Set<ConditionalOrder>>;
+
+const logger = getLogger("Registry");
```
Comment (anxolin) [inline src/types/model.ts:101]: Why registry if this is in `model.ts`?

File: src/utils/misc.ts [hunk 7]
```diff
@@ -31,3 +31,14 @@ export function toConditionalOrderParams({
...
+
+export function getAreConditionalOrderParamsEqual(
+  a: ConditionalOrderParams,
+  b: ConditionalOrderParams
+): boolean {
+  return (
+    a.handler === b.handler &&
+    a.salt === b.salt &&
+    a.staticInput === b.staticInput
+  );
```
Comment (anxolin) [inline src/utils/misc.ts:43]: Do we care about case sensitivity?

File: src/domain/events/index.ts (hunk 1 above)
Comment (shoom3301) [reply src/domain/events/index.ts:26]: I don't know, I just moved this code from the `_addContract` function in order to not create excessive instances

File: src/domain/events/index.ts (hunk 2 above)
Comment (shoom3301) [reply src/domain/events/index.ts:95]: I didn't change this code, just removed it from the exports and removed `_` suffix

File: src/domain/events/index.ts (hunk 3 above)
Comment (shoom3301) [reply src/domain/events/index.ts:215]: Added

File: src/domain/events/index.ts (hunk 4 above)
Comment (shoom3301) [reply src/domain/events/index.ts]: Thanks!
Changed it to:

```
// One increment owners and one orders
    metrics.activeOwnersTotal.labels(network).inc(2);
```

File: src/utils/misc.ts (hunk 7 above)
Comment (shoom3301) [reply src/utils/misc.ts:43]: I don't think this is a case, but can't 100% be sure.
So, I would better to lowercase them

File: src/domain/events/index.ts (hunk 1 above)
Comment (anxolin) [reply src/domain/events/index.ts:26]: I thought so. In principle are instances that are scoped to that function, so the garbage collector should be able to find (unless we have a leak in the logging library)

If you really want to do this, at least we need to find a better name than just `log`. I might want to add another function `doSomething()` and use that log, and I will see `addContract` in my log messages

File: src/domain/events/index.ts (hunk 2 above)
Comment (anxolin) [reply src/domain/events/index.ts:95]: Yes, but you created a global const with the `composableCow` instance, so no need to pass it as a parameter any more

File: src/types/model.ts (hunk 6 above)
Comment (anxolin) [reply src/types/model.ts:101]: Its a similar comment as above. You took it out of the registry and has still a very generic name. 
I don't think getting loggers should be the memory leak (i hope)

Comment (shoom3301) [reply src/types/model.ts]: Got you.
Created a separate logger:
```
const loadOwnerOrdersLogger = getLogger("loadOwnerOrders");
```

File: src/domain/events/index.ts (hunk 1 above)
Comment (shoom3301) [reply src/domain/events/index.ts]: Got you, reverted the logger back and kept only `composableCow`

**Review by anxolin — APPROVED** [review]

**Pull request merged**